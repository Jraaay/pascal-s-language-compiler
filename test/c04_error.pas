
Program example(input, output);

Var
x, y : integer;
a : record
  title: array [1..50] of char;
  author: array [1..50] of char;
  subject: array [1..100] of char;
  bookid: integer;
end;
Begin
  x := a;
End.
