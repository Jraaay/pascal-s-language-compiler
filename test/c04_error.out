{
    "ast": {
        "type": "programstruct",
        "program_head": {
            "type": "program_head",
            "ID": "example",
            "idlist": {
                "type": "idlist",
                "ids": [
                    "input",
                    "output"
                ]
            }
        },
        "program_body": {
            "type": "program_body",
            "const_declarations": null,
            "var_declarations": {
                "type": "var_declarations",
                "var_declaration": {
                    "type": "var_declaration",
                    "values": [
                        {
                            "idlist": {
                                "type": "idlist",
                                "ids": [
                                    "x",
                                    "y"
                                ]
                            },
                            "type": {
                                "type": "type",
                                "_type": {
                                    "type": "basic_type",
                                    "_type": "INTEGER"
                                }
                            }
                        },
                        {
                            "idlist": {
                                "type": "idlist",
                                "ids": [
                                    "a"
                                ]
                            },
                            "type": {
                                "type": "type",
                                "_type": "RECORD",
                                "multype": {
                                    "id": 5,
                                    "type": [
                                        {
                                            "type": "type",
                                            "_type": "ARRAY",
                                            "period": {
                                                "type": "period",
                                                "values": [
                                                    {
                                                        "start": 1,
                                                        "END": 50
                                                    }
                                                ]
                                            },
                                            "basic_type": "of"
                                        },
                                        {
                                            "type": "type",
                                            "_type": "ARRAY",
                                            "period": {
                                                "type": "period",
                                                "values": [
                                                    {
                                                        "start": 1,
                                                        "END": 50
                                                    }
                                                ]
                                            },
                                            "basic_type": "of"
                                        },
                                        {
                                            "type": "type",
                                            "_type": "ARRAY",
                                            "period": {
                                                "type": "period",
                                                "values": [
                                                    {
                                                        "start": 1,
                                                        "END": 100
                                                    }
                                                ]
                                            },
                                            "basic_type": "of"
                                        },
                                        {
                                            "type": "type",
                                            "_type": {
                                                "type": "basic_type",
                                                "_type": "INTEGER"
                                            }
                                        }
                                    ],
                                    "multype": {
                                        "id": 4,
                                        "type": [
                                            {
                                                "type": "type",
                                                "_type": "ARRAY",
                                                "period": {
                                                    "type": "period",
                                                    "values": [
                                                        {
                                                            "start": 1,
                                                            "END": 50
                                                        }
                                                    ]
                                                },
                                                "basic_type": "of"
                                            },
                                            {
                                                "type": "type",
                                                "_type": "ARRAY",
                                                "period": {
                                                    "type": "period",
                                                    "values": [
                                                        {
                                                            "start": 1,
                                                            "END": 50
                                                        }
                                                    ]
                                                },
                                                "basic_type": "of"
                                            },
                                            {
                                                "type": "type",
                                                "_type": "ARRAY",
                                                "period": {
                                                    "type": "period",
                                                    "values": [
                                                        {
                                                            "start": 1,
                                                            "END": 100
                                                        }
                                                    ]
                                                },
                                                "basic_type": "of"
                                            }
                                        ],
                                        "multype": {
                                            "id": 3,
                                            "type": [
                                                {
                                                    "type": "type",
                                                    "_type": "ARRAY",
                                                    "period": {
                                                        "type": "period",
                                                        "values": [
                                                            {
                                                                "start": 1,
                                                                "END": 50
                                                            }
                                                        ]
                                                    },
                                                    "basic_type": "of"
                                                },
                                                {
                                                    "type": "type",
                                                    "_type": "ARRAY",
                                                    "period": {
                                                        "type": "period",
                                                        "values": [
                                                            {
                                                                "start": 1,
                                                                "END": 50
                                                            }
                                                        ]
                                                    },
                                                    "basic_type": "of"
                                                }
                                            ],
                                            "multype": {
                                                "id": 2,
                                                "type": [
                                                    {
                                                        "type": "type",
                                                        "_type": "ARRAY",
                                                        "period": {
                                                            "type": "period",
                                                            "values": [
                                                                {
                                                                    "start": 1,
                                                                    "END": 50
                                                                }
                                                            ]
                                                        },
                                                        "basic_type": "of"
                                                    }
                                                ],
                                                "ID": [
                                                    "title"
                                                ]
                                            },
                                            "ID": [
                                                "title",
                                                "author"
                                            ]
                                        },
                                        "ID": [
                                            "title",
                                            "author",
                                            "subject"
                                        ]
                                    },
                                    "ID": [
                                        "title",
                                        "author",
                                        "subject",
                                        "bookid"
                                    ]
                                }
                            }
                        }
                    ]
                }
            },
            "subprogram_declarations": {
                "type": "subprogram_declarations",
                "subprograms": []
            },
            "compound_statement": {
                "type": "compound_statement",
                "statement_list": {
                    "type": "statement_list",
                    "statements": [
                        {
                            "type": "statement",
                            "_type": "variable",
                            "variable": {
                                "type": "variable",
                                "__type": "INTEGER",
                                "ID": "x",
                                "id_varpart": {
                                    "type": "id_varpart",
                                    "expression_list": null
                                }
                            },
                            "ASSIGNOP": ":=",
                            "expression": {
                                "type": "expression",
                                "__type": "RECORD",
                                "simple_expression": {
                                    "type": "simple_expression",
                                    "__type": "RECORD",
                                    "term": {
                                        "type": "term",
                                        "__type": "RECORD",
                                        "factor": {
                                            "type": "factor",
                                            "_type": "variable",
                                            "__type": "RECORD",
                                            "variable": {
                                                "type": "variable",
                                                "__type": "RECORD",
                                                "ID": "a",
                                                "id_varpart": {
                                                    "type": "id_varpart",
                                                    "expression_list": null
                                                }
                                            }
                                        }
                                    }
                                }
                            }
                        }
                    ]
                }
            }
        }
    },
    "symbolTable": {
        "constants": [],
        "variables": [
            {
                "id": 0,
                "token": "x",
                "type": "INTEGER",
                "isArray": false,
                "dimension": 0,
                "size": [],
                "start": [],
                "recordTable": null
            },
            {
                "id": 1,
                "token": "y",
                "type": "INTEGER",
                "isArray": false,
                "dimension": 0,
                "size": [],
                "start": [],
                "recordTable": null
            },
            {
                "id": 6,
                "token": "a",
                "type": "RECORD",
                "isArray": false,
                "dimension": 0,
                "size": [],
                "start": [],
                "recordTable": {
                    "variables": [
                        {
                            "id": 2,
                            "token": "title",
                            "type": "CHAR",
                            "isArray": true,
                            "dimension": 1,
                            "size": [
                                50
                            ],
                            "start": [
                                1
                            ],
                            "recordTable": null
                        },
                        {
                            "id": 3,
                            "token": "author",
                            "type": "CHAR",
                            "isArray": true,
                            "dimension": 1,
                            "size": [
                                50
                            ],
                            "start": [
                                1
                            ],
                            "recordTable": null
                        },
                        {
                            "id": 4,
                            "token": "subject",
                            "type": "CHAR",
                            "isArray": true,
                            "dimension": 1,
                            "size": [
                                100
                            ],
                            "start": [
                                1
                            ],
                            "recordTable": null
                        },
                        {
                            "id": 5,
                            "token": "bookid",
                            "type": "INTEGER",
                            "isArray": false,
                            "dimension": 0,
                            "size": [],
                            "start": [],
                            "recordTable": null
                        }
                    ]
                }
            }
        ],
        "subFunc": []
    },
    "error": [
        {
            "code": "C-04",
            "info": {
                "line": 11,
                "value": [
                    "x",
                    "INTEGER",
                    "RECORD"
                ],
                "lexpos": 204
            }
        }
    ],
    "warning": []
}